### PR TITLE
Instrument spring-web 6 & spring-webmvc 6

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Part of #7203

This PR is mostly copy-paste and working around the differences, conceptually the new instrumentation is the same as the old one

### CODE REVIEW

**Review by laurit — APPROVED** [review]

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/DispatcherServletInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  public static class HandlerMappingAdvice {
+
+    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
+    public static void afterRefresh(
+        @Advice.Argument(0) ApplicationContext springCtx,
+        @Advice.FieldValue("handlerMappings") List<HandlerMapping> handlerMappings) {
+      if (springCtx.containsBean("otelAutoDispatcherFilter")) {
+        OpenTelemetryHandlerMappingFilter filter =
+            (OpenTelemetryHandlerMappingFilter) springCtx.getBean("otelAutoDispatcherFilter");
+        if (handlerMappings != null) {
```
Comment (trask) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/DispatcherServletInstrumentation.java:67]: what was reason for not keeping the `filter != null` condition?

**Review by trask — APPROVED** [review]
do you think we should port `ServletFilterTest` to 6.0 at some point?

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/DispatcherServletInstrumentation.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/DispatcherServletInstrumentation.java:67]: `getBean()` will never return null - it'll either return the filter or throw.

File: instrumentation/spring/spring-webmvc/spring-webmvc-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/HandlerSpanNameExtractor.java [hunk 2]
```diff
@@ -44,4 +43,10 @@ public String extract(Object handler) {
 
     return SpanNames.fromMethod(clazz, methodName);
   }
+
+  private static boolean isServlet(Object handler) {
+    String handlerClassName = handler.getClass().getName();
+    return handlerClassName.equals("javax.servlet.Servlet")
+        || handlerClassName.equals("jakarta.servlet.Servlet");
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply instrumentation/spring/spring-webmvc/spring-webmvc-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/HandlerSpanNameExtractor.java:50]: Great catch! Will fix that 👍

**Comment by mateuszrzeszutek** [comment]
> do you think we should port `ServletFilterTest` to 6.0 at some point?

Yeah, we should. I didn't want to do it in this PR, as it's already quite huge. I'll add the test in one of the next PRs.

**Pull request merged**